File: Iteration 1/CSE3063F24P1_GRP14/src/CourseSection.java
import com.fasterxml.jackson.annotation.JsonProperty;

class CourseSection {

    @JsonProperty("day")
    private String day;

    @JsonProperty("place")
    private String place;

    @JsonProperty("hour")
    private String hour;

    @JsonProperty("year")
    private int year;

    @JsonProperty("instructor")
    private String instructor;

    @JsonProperty("enrollmentCapacity")
    private int enrollmentCapacity;

    @JsonProperty("status")
    private String status;

    // Getters
    public int getEnrollmentCapacity() {
        return enrollmentCapacity;
    }

    public CourseSection() {}


    public CourseSection(String sanemArslan, int i, String active) {
        day = sanemArslan;
    }

    // toString method for printing
    @Override
    public String toString() {
        return "CourseSection{" +
                "place='" + place + '\'' +
                ", day='" + day + '\'' +
                ", hour='" + hour + '\'' +
                ", year=" + year +
                ", instructor='" + instructor + '\'' +
                ", enrollmentCapacity=" + enrollmentCapacity +
                ", status='" + status + '\'' +
                '}';
    }

}

File: Iteration 3/src/course_section.py
class CourseSection:
    def __init__(self, day: str = "", place: str = "", hour: str = ""):
        self.day = day
        self.place = place
        self.hour = hour

    # Getter methods
    def getDay(self) -> str:
        return self.day

    def getPlace(self) -> str:
        return self.place

    def getHour(self) -> str:
        return self.hour

    # Utility method
    def __str__(self) -> str:
        return f"Day: {self.day}, Place: {self.place}, Hour: {self.hour}"

